templates/fine-tune-function-calling/README.md

```diff
@@ -0,0 +1,419 @@
+1. Fine-tuning on the Anyscale Platform through our fine-tuning template 
+    - This would be the preferred route for those wishing to get more flexibility in choice of models and hyperparameters, better monitoring, etc.
+2. Fine-tuning through Anyscale's serverless endpoints
+    - A quick and easy way to fine-tune a model via an OpenAI compatiable SDK.
+
+For this guide, we will use `Llama-3-8B-Instruct` as the base model for fine-tuning.
+
+
+
+## Step 2(a): Fine-tuning on the Anyscale Platform
+
+Head over to the Anyscale Platform: https://console.anyscale.com/v2 and spin up the "Fine-tune LLMs" template (under "AI application templates")
+
+<p align="center">
+  <img src="./assets/templates.png" alt="Templates">
+</p>
+
+
+
+Follow the instructions to run your fine-tuning job.
+
+## Step 2(b): Fine-tuning through serverless endpoints
+First, obtain your credentials from the [Anyscale platform](https://console.anyscale.com/v2/) and upload the training and validation files in the [fine-tuning tab](https://console.anyscale.com/v2/fine-tuning?fine-tuning-tab=files). Make a note of the file IDs for each. This will be passed to the fine-tuning job.
+
+
+```python
+ANYSCALE_API_KEY = "esecret_yourKeyHere"  # from https://console.anyscale.com/credentials
+ANYSCALE_API_BASE = "https://api.endpoints.anyscale.com/v1"
+```
+
+
+```python
+# Anyscale Endpoints are OpenAI compatible
+client = openai.OpenAI(
+    base_url = ANYSCALE_API_BASE,
+    api_key = "esecret_yourKeyHere" 
+)
+```
+
+Let's now launch a fine-tuning job for 4 epochs. The expected time for this job is < 3 hours.
+
+
+```python
+# other parameters like context length will be chosen appropriately based on dataset size
+client.fine_tuning.jobs.create(
+    model="meta-llama/Meta-Llama-3-8B-Instruct",
+    hyperparameters={"n_epochs": 4},
+    # replace with the actual file ids!
+    training_file="file_trainingFileId",
+    validation_file="file_validationFileId",
+)
+```
+
+# Step 3: Serving
+
+## Step 3(a): Finetuned on the Anyscale Platform
+
+Make a note of the final checkpoint after fine-tuning (this should be the last line in the logs). You can now spin up the "Deploy LLMs" template which has all the instructions and required dependencies to serve your finetuned model efficiently. You will find the tutorials on [serving LoRA models](https://github.com/anyscale/templates/blob/main/templates/endpoints_v2/examples/lora/DeployLora.ipynb) (if applicable) and on deploying a [custom model](https://github.com/anyscale/templates/blob/main/templates/endpoints_v2/examples/CustomModels.ipynb) helpful. Once you have set up your fine-tuned model as an Anyscale Service, note down the base URL and API key and place them here.
+
+
+```python
+# to be run only if you finetuned on the Anyscale platform
+ANYSCALE_API_KEY="service-api-key-here"
+ANYSCALE_API_BASE="service-url-here" 
+if not ANYSCALE_API_BASE.endswith("/"):
+    ANYSCALE_API_BASE += "/"
+ANYSCALE_API_BASE += "v1"
```

Comment:
use fstring

Reply:
just comment on what they should paste (don't use the endswith `/` stuff.) Maybe a screen shot with highlights around the text that is important.

Reply:
Yup added.